jetstream/bigquery_client.py

```diff
@@ -185,18 +183,17 @@ def delete_experiment_tables(
         self, slug: str, analysis_periods: list[AnalysisPeriod], delete_enrollments: bool = False
     ):
         """Delete all tables associated with the specified experiment slug."""
-        normalized_slug = bq_normalize_name(slug)
         analysis_periods_re = "|".join([p.value for p in analysis_periods])
+        analysis_periods_pattern = re.compile(f"_({analysis_periods_re}).*$")
 
-        existing_tables = self.tables_matching_regex(
-            f"^{normalized_slug}_.+_({analysis_periods_re}).*$"
-        )
-        existing_tables += self.tables_matching_regex(
-            f"^statistics_{normalized_slug}_({analysis_periods_re}).*$"
-        )
+        enrollments_table = f"enrollments_{bq_normalize_name(slug)}"
 
-        if delete_enrollments:
-            existing_tables += self.tables_matching_regex(f"^enrollments_{normalized_slug}$")
+        existing_tables = [
+            table_name
+            for table_name in self.tables_matching_description(slug)
+            if (delete_enrollments and table_name == enrollments_table)
+            or analysis_periods_pattern.search(table_name) is not None
```

Comment:
I'm uncertain about whether these are the right way to encode these conditions, but I guess it does at least closely match existing behavior. happy to change if you have something else in mind here.

Reply:
Oh nevermind, I see we need to filter to the specified analysis periods... In that case what you have looks good!